examples/bank_account/bank_account.py

```diff
@@ -26,7 +26,7 @@ def balance(self):
     def withdraw(self, amount):
         """ withdraw money """
         if self._balance >= amount > 0:
-            self._balance -= (amount + self._calc_commission_rate(self._hasCommissionDiscount))
+            self._balance += (amount + self._calc_commission_rate(self._hasCommissionDiscount))
```

Comment:
**Suggestion:** The balance update operation in the withdraw method seems to be incorrect. The balance should decrease when a withdrawal is made, not increase.
```suggestion
    def withdraw(self, amount):
        """ withdraw money """
        if self._balance >= amount > 0:
            self._balance -= (amount + self._calc_commission_rate(self._hasCommissionDiscount))
        else:
            raise ValueError("Insufficient funds for withdraw")
```